Update on README, UserGuide (#5)
openshift/kube-compare: docs/UserGuide.md

# Example use cases

To Compare a known valid reference configuration with a live cluster:

`kubectl cluster-compare -r <referenceConfigurationDirecotry>`

To Compare a known valid reference configuration with a local set of CRs:

`kubectl cluster-compare -r <referenceConfigurationDirecotry> -f <inputConfiguration>`

To Compare a known valid reference configuration with a live cluster and with a user config:

`kubectl cluster-compare -r <referenceConfigurationDirecotry> -c <userConfig>`

To Run a known valid reference configuration with a must-gather output:

`kubectl cluster-compare -r <referenceConfigurationDirecotry> -f "must-gather*/*/cluster-scoped-resources","must-gather*/*/namespaces" -R`

# Understanding the output

# Options and advanced usage

## Kubectl Environment Variables
The tool is responsive to KUBECTL_EXTERNAL_DIFF environment variable (same as oc diff). This allows you to tailor the output formatting to suit your preference. 

side-by-side comparison (total width 150 characters) with:

> `KUBECTL_EXTERNAL_DIFF="diff -y -W 150"`

Reviewer: I think we coud explain what each option individually achieves.
-y : side by side comparison
--color: colored output
-W <char_wifth>: width of char_length
And then say. For more available options do `diff --help`

# Troubleshooting

## False Positives

### Reference Designs CRs may change (or not) from one cluster version to another version
The tool sometimes reports CR to be missing completely while in the live cluster environment the CR is present. This can happen at times because the tool is not correctly identifying the operator versions of these CRs or template's critical fields don't match cluster CRs.

#### Example #1:
```
Missing required CRs: 
DU-Reference:
  Tuning:
  - ConsoleOperatorDisable.yaml
  - DisableSnoNetworkDiag.yaml

[cluster]$ kubectl get network -A
NAME      AGE
cluster   5d20h
[cluster]$ kubectl get console -A
NAME      AGE
cluster   5d20h
```
#### Example #2:
```
Cluster CR
apiVersion: sriovnetwork.openshift.io/v1
kind: SriovNetworkNodePolicy
metadata:
  name: $name
  namespace: openshift-sriov-network-operator
  
Template CR
apiVersion: sriovnetwork.openshift.io/v1
kind: SriovNetworkNodePolicy
metadata:
  name: sriov

>   namespace: openshift-sriov-network-operator

Reviewer: Suggest to not use telco keywords like DU-Reference, Tuning for the wider audience, . You can use terms like "ExamplePart" "ExampleComponent".

# Example use cases

To Compare a known valid reference configuration with a live cluster:

`kubectl cluster-compare -r <referenceConfigurationDirecotry>`

To Compare a known valid reference configuration with a local set of CRs:

`kubectl cluster-compare -r <referenceConfigurationDirecotry> -f <inputConfiguration>`

> To Compare a known valid reference configuration with a live cluster and with a user config:

Reviewer 2: same as above

`kubectl cluster-compare -r <referenceConfigurationDirecotry> -c <userConfig>`

> To Run a known valid reference configuration with a must-gather output:

Reviewer 2: same as above

`kubectl cluster-compare -r <referenceConfigurationDirecotry> -f "must-gather*/*/cluster-scoped-resources","must-gather*/*/namespaces" -R`

# Understanding the output

# Options and advanced usage

## Kubectl Environment Variables

> The tool is responsive to KUBECTL_EXTERNAL_DIFF environment variable (same as oc diff). This allows you to tailor the output formatting to suit your preference. 

Reviewer 2: ```suggestion
The tool is responsive to KUBECTL_EXTERNAL_DIFF environment variable (same as kubectl diff). This allows you to tailor the output formatting to suit your preference. 
```

> -y : side by side comparison

Reviewer 2: ```suggestion
For example, with the standard diff utility:
-y : side by side comparison
```

For more available options do diff --help

side-by-side comparison (total width 150 characters) with:
`KUBECTL_EXTERNAL_DIFF="diff -y -W 150"`

# Troubleshooting

## False Positives

### Reference Designs CRs may change (or not) from one cluster version to another version
The tool sometimes reports CR to be missing completely while in the live cluster environment the CR is present. This can happen at times because the tool is not correctly identifying the operator versions of these CRs or template's critical fields don't match cluster CRs.

#### Example #1:
```
Missing required CRs: 
Template-Reference:
  ExamplePart:
  - Example-Component-CR.yaml
  - Example-Component-CR.yaml

[cluster]$ kubectl get example-component -A
NAME                      AGE
example-component-name   5d20h
[cluster]$ kubectl get console -A
NAME                      AGE
example-component-name   5d20h
```
#### Example #2:
```
Cluster CR
apiVersion: ExampleVersion
kind: ExampleKind
metadata:
  name: $x-name
  namespace: ExampleNamespace
  
Template CR
apiVersion: ExampleVersion
kind: ExampleKind
metadata:
  name: $y-name
  namespace: ExampleNamespace
```

In such scenarios take these steps:

> 1. Ensure the lastest version of tool is updated with the most recent version of CR.

Reviewer 2: ```suggestion
1. Ensure you are using the lastest version of tool
1. Ensure you are using the most up-to-date version of the reference configuration.
```

2. Ensure that template has the same api-version-kind-name-namespace as the cluster CR.

### There can be more than one Reference Design CR of the same Kind

In this case you will have a warning presented before the diff output, formatted similar to this:
`W0402 15:19:36.531169 3578413 corelator.go:144] More then one template with same apiVersion, metadata_name, metadata_namespace, kind. These templates wont be used for corelation. To use them use different corelator (manual matching) or remove one of them from the reference. Template names are: XConfig.yaml, YConfig.yaml`

> This means the template contains two CRs with the same apiversion-kind-name-namespace but different spec. In such cases comment out the template CRs from the metadata.yaml that are more accurate with CRs under comparison.

Reviewer 2: You can also specify a manual correlation in the user config. @AlinaSecret do you have an example?
